engine/consensus/Sealing_Readme.md

```diff
@@ -0,0 +1,109 @@
+    - Lastly, the builder adds receipts from any valid block that descends from the latest finalized block.
+    - Note:
+
+      It is *strictly required* to add the result for the latest block that is both sealed and finalized. This is necessary because the ID of the latest result is given as a query to find all descending results during block construction. Without the sealed result, the block builder could not proceed.
+
+      In contrast, conceptually it is *not strictly required* for the builder to also add the receipts from the known blocks (finalized unsealed blocks and all pending blocks descending from the latest finalized block). This is because this matching engine *would* re-request these (method `requestPendingReceipts`) by itself. Nevertheless, relying on re-feting already known receipts would add a large latency overhead after recovering from a crash.
+
+
+### Sealing Engine
+
+- The sealing engine ingests:
+    - Result approvals from the Verification nodes (message `ResultApproval`)
+    - It also re-requests missing result approvals and ingests the corresponding responses (`ApprovalResponse`)
+    - Execution Results from incorporated blocks. Sealing engine can trust those incorporated results are structurally valid, because block incorporating the result has passed the node's internal compliance engine.
+
+      Caution: to not compromise sealing liveness, the sealing engine cannot drop any incorporated blocks or any results contained in it. Otherwise, it cannot successfully create a seal for the incorporated results.
+
+- Sealing Engine will forward receipts and approvals to the sealing core. Receipts and approvals are concurrently processed by sealing core.
+- The purpose for the Sealing Core is to
+    1. compute verifier assignments for the incorporated results
+    2. track the approvals that match the assignments
+    3. generate a seal once enough approvals have been collected
+- For sealing core to concurrently process receipts, it maintains an `AssignmentCollectorTree`, which is a fork-aware structure holding the assignments + approvals for each incorporated result. Therefore, it can ingest results and approvals in arbitrary order.
+    - Each vertex (`AssignmentCollector`) in the tree is responsible for managing approvals for one particular result.
+    - A result might be incorporated in multiple blocks on different block forks. 
+    - Results incorporated on different forks will lead to different assignment for verification nodes. For example, consider block `B` with two children `C` and `D`:
+      ```
+            ┌ C   
+         B <   
+            └ D
+      ```
+      `C` and `D` can each incorporate the same result `r` for block `B`, but the verifier assignment would be different. Hence, there is one `AssignmentCollector` for `r` stored in the `AssignmentCollectorTree`.
```

Comment:
Can we show on diagram that `C` and `D` incorporates `ER` for `B`?

Reply:
Thanks for this comment. I realized that there is quite a bit of information missing, to properly explain what is going on here. I am worried that adding additional context here to this figure, and using notation that I have to introduce might be too confusing. 

* I added two additional sections (exported from our notion documents) to explain more of the basics upfront ... also including a proper figure that illustrates this example a bit more in detail. 
* Instead of adding more complexity here, I could then simply just reference the example from the prior section.